src/metabase/models/emitter.clj

```diff
@@ -51,23 +66,21 @@
   (db/delete! Emitter :id (:emitter_id emitter))
   emitter)
 
-(u/strict-extend
-  (class DashboardEmitter)
-  models/IModel
+(def ^:private Emitter-subtype-IModel-impl
+  "[[models/IModel]] impl for `DashboardEmitter` and `CardEmitter`"
   (merge models/IModelDefaults
-         {:primary-key (constantly :emitter_id) ;; This is ok as long as we're 1:1
+         {:primary-key (constantly :emitter_id) ; This is ok as long as we're 1:1
           :pre-delete pre-delete
           :pre-update pre-update
           :pre-insert pre-insert}))
 
-(u/strict-extend
-  (class CardEmitter)
+(u/strict-extend (class DashboardEmitter)
   models/IModel
-  (merge models/IModelDefaults
-         {:primary-key (constantly :emitter_id) ;; This is ok as long as we're 1:1
-          :pre-delete pre-delete
-          :pre-update pre-update
-          :pre-insert pre-insert}))
+  Emitter-subtype-IModel-impl)
+
+(u/strict-extend (class CardEmitter)
+  models/IModel
+  Emitter-subtype-IModel-impl)
```

Comment:
I deduplicated this a bit so we don't need to specify the exact same method map twice